### PR TITLE
gh-125887: Update PyObject_HasAttr exception behavior

### LINKED ISSUE #125887: Outdated PyObject_HasAttr documentation (Exception behavior)
# Documentation

In Python 3.13, `PyObject_HasAttr` doc says:

> Exceptions that occur when this calls `__getattr__()` and `__getattribute__()` methods are silently ignored.

But its exception behavior after https://github.com/python/cpython/pull/106674 is to report all exceptions other than `AttributeError`.

### Linked PRs
* gh-125907
* gh-128283

### PR DESCRIPTION
**Pull request by Damien-Chen**
Resolve #125887 

* Issue: gh-125887

----
📚 Documentation preview 📚: https://cpython-previews--125907.org.readthedocs.build/

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
I think we should continue to mention the alternatives

File: Doc/c-api/object.rst [hunk 1]
```diff
@@ -111,9 +111,8 @@ Object Protocol
    .. note::
 
       Exceptions that occur when this calls :meth:`~object.__getattr__` and
-      :meth:`~object.__getattribute__` methods are silently ignored.
-      For proper error handling, use :c:func:`PyObject_HasAttrWithError`,
-      :c:func:`PyObject_GetOptionalAttr` or :c:func:`PyObject_GetAttr` instead.
+      :meth:`~object.__getattribute__` methods are aren't propagated,
+      but instead given to :func:`sys.unraisablehook`.
```
Comment (skirpichev) [inline Doc/c-api/object.rst]:
```suggestion
      :meth:`~object.__getattribute__` methods aren't propagated,
      but instead given to :func:`sys.unraisablehook`.
      For proper error handling, use :c:func:`PyObject_HasAttrWithError`,
      :c:func:`PyObject_GetOptionalAttr` or :c:func:`PyObject_GetAttr` instead.      
```

**Review by skirpichev — APPROVED** [review]

**Comment by skirpichev** [comment]
@Damien-Chen, please avoid force-pushes.

**Comment by Damien-Chen** [comment]
I want to rebase and squash my commit, but `--force push `seems necessary.

**Pull request merged**